src/args/make-node-js-modules/default.nix

```diff
@@ -22,6 +23,7 @@
   collectDependencies = deps:
     builtins.foldl'
     (all: name: let
+      registryToken = registryTokens.${name} or null;
```

Comment:
I'm wondering if it's better to use `builtins.getEnv` so it can be paired with `secretsForEnvFromSops`

Reply:
Tokens are secrets, So I guess we need to handle them as envars.

Reply:
is it possible to declare `secretsForEnvFromSops` and `makeNodeJsModules` in the same main.nix?

Reply:
Yes, you can use `searchPaths.source`, this will basically execute the script provided by `secretsForEnvFromSops`.

Reply:
I'm having a hard time declaring `secretsForEnvFromSops` in the same file

```
{
  makeNodeJsModules,
  projectPath,
  secretsForEnvFromSops,
  ...
}: let
 secretsForEnvFromSops = {
    test = {
      vars = ["FONTAWESOME_PRO_TOKEN"];
      manifest = "/test/dev.yaml";
    };
  };
in
makeNodeJsModules {
  name = "test";
  nodeJsVersion = "18";
  packageJson = projectPath "/test/package.json";
  packageLockJson = projectPath "/test/package-lock.json";
  registryTokens = {
    "@fortawesome/fontawesome-pro" = "FONTAWESOME_PRO_TOKEN";
  };
  searchPaths.source = [secretsForEnvFromSops.test];
}
```

Would I have to wrap the expression? soz, I'm fairly new actually building stuff with the language

Reply:
nvm, I figured it out. Had to use `makeSecretForEnvFromSops` instead of `secretsForEnvFromSops`